src/policy/fees.cpp

```diff
@@ -580,13 +581,20 @@ void CBlockPolicyEstimator::TransactionRemovedFromMempool(const CTransactionRef
     removeTxNotInBlock(tx->GetHash());
 }
 
-bool CBlockPolicyEstimator::processBlockTx(unsigned int nBlockHeight, const uint256& hash)
+bool CBlockPolicyEstimator::processBlockTx(unsigned int nBlockHeight, const CTransactionRef& tx)
 {
-    std::map<uint256, TxStatsInfo>::iterator pos = mapMemPoolTxs.find(hash);
+    std::map<uint256, TxStatsInfo>::iterator pos = mapMemPoolTxs.find(tx->GetHash());
     if (pos == mapMemPoolTxs.end()) {
         // This transaction wasn't being tracked for fee estimation
         return false;
     }
+    if (pos->second.witness_hash != tx->GetWitnessHash()) {
+        // In the case that we saw a tx which had its witness malleated,
```

Comment:
commit: Track witness hash malleation in fee estimator

Why don't we want to record the blocks to confirm in this case?

Reply:
Because we don't know how long its been around/it waited.

Reply:
I don't understand. We have the record of when the tx entered the mempool (albeit with different witness data). How does the malleated witness change things?

Reply:
Because clearly there was a propagation issue. We dont know whether the lower-feerate version (or some other version) was the one most miners saw or the higher-feerate version. Better to just ignore it.